### PR TITLE
[Fluent2 iOS] Fluent2 BadgeLabel

### PR DESCRIPTION
**Pull request by edjamesmsft**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Binary change:

| Before | After |
|--------|-------|
| 27,427,169 | 27,429,089 |

Updating BadgeLabel to be Fluent2 compliant. This indirectly affects NavigationBar and TabBar.

### Verification

Built and ran simulator, used Colour Contrast Analyser to ensure colors were as expected and also screen shots looked correct.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 14 Pro - 2022-12-02 at 14 54 52 | Simulator Screen Shot - iPhone 14 Pro - 2022-12-02 at 15 34 05
vbar light |
| Simulator Screen Shot - iPhone 14 Pro - 2022-12-02 at 14 40 14 | Simulator Screen Shot - iPhone 14 Pro - 2022-12-02 at 15 34 13 |
| Simulator Screen Shot - iPhone 14 Pro - 2022-12-02 at 15 44 33 | Simulator Screen Shot - iPhone 14 Pro - 2022-12-02 at 15 33 47 |
| Simulator Screen Shot - iPhone 14 Pro - 2022-12-02 at 15 44 37 | Simulator Screen Shot - iPhone 14 Pro - 2022-12-02 at 15 33 52 |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1426)

### CODE REVIEW

File: ios/FluentUI/Label/BadgeLabel.swift [hunk 1]
```diff
@@ -23,21 +23,26 @@ class BadgeLabel: UILabel {
...
-        backgroundColor = Colors.Palette.dangerPrimary.color
-        textColor = .white
         textAlignment = .center
         font = UIFont.systemFont(ofSize: Constants.badgeFontSize, weight: .regular)
         isHidden = true
+
+        updateColors()
     }
 
     override func didMoveToWindow() {
```
Comment (laminesm) [inline ios/FluentUI/Label/BadgeLabel.swift:33]: Do we need this function anymore since we're not using the window to update colors?

Comment (huwilkes) [reply ios/FluentUI/Label/BadgeLabel.swift:33]: Different windows can have different fluent themes, so the colors might still change between windows. So I think we should still have this, and we would probably just add it back when we tokenize the control.

Comment (edjamesmsft) [reply ios/FluentUI/Label/BadgeLabel.swift:33]: Agree with huwilkes - I think it's worth keeping this function!

Comment (edjamesmsft) [reply ios/FluentUI/Label/BadgeLabel.swift:38]: The compromise is to add themeDidChange alongside didMoveToWindow.

**Review by huwilkes — APPROVED** [review]

**Review by laminesm — APPROVED** [review]

**Pull request merged**